ggshield/core/types.py

```diff
@@ -1,11 +1,14 @@
-from dataclasses import dataclass
 from typing import Optional
 
 import marshmallow_dataclass
+from marshmallow import EXCLUDE
 
 
-@dataclass
+@marshmallow_dataclass.dataclass
 class IgnoredMatch:
+    class Meta:
+        unknown = EXCLUDE
```

Comment:
This is going to silently ignore unknown keys in matches. I think you can fix this by moving BaseConfig here so that IgnoredMatch can use it.

Also, can you rename BaseConfig to something like FilteredConfig?

Reply:
I know, I was wondering if we wanted to alert in that case:
- extra fields won't change anything in the behaviour of ggshield (typos in the field name as well)
- it can result in a lot of warning if an extra field is present with too many occurences (if you ignore 100 secrets, you get 100 warnings)
- some users may want to add extra information to the ignored match and it can be useful to allow other keys
- it is the current behaviour in the main branch
- 
On the other hand,
- It is a different behaviour than the remaining config
- it may be harder to spot a typo in the "match" key since it will only show an error on a missing key but not the extra one.

WDYT ?

Reply:
Ok, I have done it.
It must be noted that this changes the behaviour of V1 config where it will now alert on extra keys.